src/include/sof/audio/pcm_converter.h

```diff
@@ -27,7 +27,7 @@ struct audio_stream;
 #endif
 
 #ifndef UNIT_TEST
-#if XCHAL_HAVE_HIFI3 && CONFIG_FORMAT_CONVERT_HIFI3
+#if __XCC__ && XCHAL_HAVE_HIFI3 && CONFIG_FORMAT_CONVERT_HIFI3
```

Comment:
Wait, so GCC can define XCHAL_HAVE_HIFI3 even though it won't emit proper code?

Reply:
`XCHAL_HAVE_HIFI3` id core property, which is toolchain independent, but GCC can't properly handle intrinsic HiFi3 instructions